reports/filesDataExporter/batch/batch_test.go

```diff
@@ -0,0 +1,309 @@
+	}
+	bucket2 = &models.BucketDescriptor{
+		Name:    "Bucket2",
+		Created: time1,
+	}
+	file1V1 = &models.FileDescriptor{
+		Archetype:   "openEHR-EHR-OBSERVATION.blood_pressure.v1",
+		Checksum:    "CHS",
+		ContentType: "text/openEhrXml",
+		Created:     time1,
+		Name:        "File1",
+		Path:        "Bucket1/File1/V1",
+		Version:     "V1",
+		Size:        8,
+		Operation:   "w",
+	}
+	file1V2 = &models.FileDescriptor{
+		Archetype:   "openEHR-EHR-OBSERVATION.blood_pressure.v1",
+		Checksum:    "CHS",
+		ContentType: "text/openEhrXml",
+		Created:     time4,
+		Name:        "File1",
+		Path:        "Bucket1/File1/V2",
+		Version:     "V2",
+		Size:        8,
+		Operation:   "w",
+	}
+	file1V3 = &models.FileDescriptor{
+		Archetype:   "openEHR-EHR-OBSERVATION.blood_pressure.v1",
+		Checksum:    "CHS",
+		ContentType: "text/openEhrXml",
+		Created:     time5,
+		Name:        "File1",
+		Path:        "Bucket1/File1/V3",
+		Version:     "V3",
+		Size:        8,
+		Operation:   "d",
+	}
+	file2V1 = &models.FileDescriptor{
+		Archetype:   "",
+		Checksum:    "CHS",
+		ContentType: "image/jpeg",
+		Created:     time1,
+		Name:        "File2",
+		Path:        "Bucket1/Image/V1",
+		Version:     "V1",
+		Size:        15698,
+		Operation:   "w",
+	}
+	file2V2 = &models.FileDescriptor{
+		Archetype:   "",
+		Checksum:    "CHS",
+		ContentType: "image/jpeg",
+		Created:     time3,
+		Name:        "File2",
+		Path:        "Bucket1/Image/V2",
+		Version:     "V2",
+		Size:        0,
+		Operation:   "d",
+	}
+	file3V1 = &models.FileDescriptor{
+		Archetype:   "openEHR-EHR-OBSERVATION.body_mass_index.v2",
+		Checksum:    "CHS",
+		ContentType: "text/openEhrXml",
+		Created:     time1,
+		Name:        "File3",
+		Path:        "Bucket2/File3/V1",
+		Version:     "V1",
+		Size:        8,
+		Operation:   "w",
+	}
+	file3V2 = &models.FileDescriptor{
+		Archetype:   "openEHR-EHR-OBSERVATION.body_mass_index.v2",
+		Checksum:    "CHS",
+		ContentType: "text/openEhrXml",
+		Created:     time2,
+		Name:        "File3",
+		Path:        "Bucket2/File3/V2",
+		Version:     "V2",
+		Size:        8,
+		Operation:   "w",
+	}
+	file3V3 = &models.FileDescriptor{
+		Archetype:   "openEHR-EHR-OBSERVATION.body_mass_index.v2",
+		Checksum:    "CHS",
+		ContentType: "text/openEhrXml",
+		Created:     time4,
+		Name:        "File3",
+		Path:        "Bucket3/File3/V3",
+		Version:     "V3",
+		Size:        8,
+		Operation:   "w",
+	}
+	noErrors   = false
+	withErrors = true
+)
+
+func TestExport(t *testing.T) {
+	testCases := []struct {
+		description   string
+		lastRun       time.Time
+		mockCalls     func(*mock.MockHandlers) []*gomock.Call
+		errorExpected bool
+		exactError    error
+	}{
+		{
+			"Export succesful",
+			time.Time(time3),
+			func(c *mock.MockHandlers) []*gomock.Call {
+				return []*gomock.Call{
+					c.EXPECT().
+						ListSourceBuckets(gomock.Any()).
+						Return([]*models.BucketDescriptor{bucket1, bucket2}, nil).
+						Times(1),
+					c.EXPECT().
```

Comment:
invalid operation: c (variable of type *invalid type) has no field or method EXPECT